common/concurrency/limits/src/main/java/io/helidon/common/concurrency/limits/LimitAlgorithm.java

```diff
@@ -41,54 +41,126 @@ public interface LimitAlgorithm {
+     * <li>Make sure the {@code limitOutcomeConsumer} is non-null, and after determining the disposition of the item of work
+     * create a suitable {@code LimitOutcome} and pass it to the consumer. Also, make sure to use an outcome-aware token
+     * internally so when the caller invokes the token's methods the outcome is updated accordingly.</li>
+     * </ul>
+     *
+     * @param callable                 callable to execute within the limit
+     * @param <T>                      the callable return type
+     * @return result of the callable with the outcome
+     * @throws LimitException      in case the limiter did not have an available permit
+     * @throws java.lang.Exception in case the task failed with an exception
+     */
+    default <T> Result<T> call(Callable<T> callable) throws Exception {
+        Outcome outcome = tryAcquireOutcome();
+        if (outcome instanceof Outcome.Accepted accepted) {
+            var permit = accepted.token();
+            try {
+                T response = callable.call();
+                permit.success();
+                return Result.create(response, outcome);
+            } catch (IgnoreTaskException e) {
+                permit.ignore();
+                return Result.create(e.handle(), outcome);
+            } catch (Exception e) {
+                permit.dropped();
+                throw e;
+            }
         }
+        throw new LimitException("No token available.");
     }
 
     /**
      * Invoke a runnable within the limits of this limiter.
      * <p>
-     * {@link io.helidon.common.concurrency.limits.Limit} implementors note:
+     * {@link io.helidon.common.concurrency.limits.Limit} implementor's note:
      * Make sure to catch {@link io.helidon.common.concurrency.limits.IgnoreTaskException} from the
      * runnable, and call its {@link IgnoreTaskException#handle()} to either return the provided result,
      * or throw the exception after ignoring the timing for future decisions.
      *
      * @param runnable runnable to execute within the limit
      * @throws LimitException      in case the limiter did not have an available permit
      * @throws java.lang.Exception in case the task failed with an exception
+     * @deprecated Use {@link #run(Runnable)}.
      */
-    default void invoke(Runnable runnable) throws LimitException, Exception {
-        Optional<Token> token = tryAcquire();
-        if (token.isEmpty()) {
-            throw new LimitException("No token available.");
-        }
-        Token permit = token.get();
-        try {
-            runnable.run();
-            permit.success();
-        } catch (IgnoreTaskException e) {
-            permit.ignore();
-            e.handle();
-        } catch (Exception e) {
-            permit.dropped();
-            throw e;
+    @Deprecated(since = "4.3.0", forRemoval = true)
+    void invoke(Runnable runnable) throws LimitException, Exception;
```

Comment:
This is a backward incompatible change - method was `default`, must stay `default`